### PR TITLE
Ha 24: api update note and mer HA-28

### PR DESCRIPTION
**Pull request by ThaiHaiDev**

### CODE REVIEW

**Comment by trunghieu0207** [comment]
https://github.com/cvn-intern/internship_22_HoiAn_smart_note/pull/15/files#diff-5ae09c022e70b52086a7c66d602a88eb78903a33b92e8a3c86878f47e4cdb279
Revert lại file này!

File: smart-note-client/src/Sidebar/Navigation/Navigation.module.scss [hunk 1]
```diff
@@ -1,7 +1,13 @@
+.root {
+    background: black !important;
+}
 .menu {
     background: #0076ff;
     color: white;
     border-radius: 20px 20px 0 0 ;
+    margin: 0 25px 0 25px;
+    background-color: #0076ff;
+    // style={{margin: '0 25px 0 25px', width: 'auto', background: '#328cff', border: 'black'}}
```
Comment (bao8816) [inline smart-note-client/src/Sidebar/Navigation/Navigation.module.scss:10]: Deadcode?

File: smart-note-server/app/Http/Controllers/NoteController.php [hunk 2]
```diff
@@ -5,85 +5,85 @@
...
+        } catch (Exception $exception) {
             return response()->json([
                 'error' => $exception->getMessage(),
-            ], 500);
+            ], StatusCode::$internalServer);
         }
     }
 
-    public function readNote($note_id)
+    public function getNote(string $note_id)
```
Comment (bao8816) [inline smart-note-server/app/Http/Controllers/NoteController.php:33]: getNote() and getNotes(): easy to confuse

File: smart-note-server/app/Http/Controllers/NoteController.php [hunk 3]
```diff
@@ -5,85 +5,85 @@
...
+            if (!isset($validCategory)) {
                 return response()->json([
                     'message' => 'Category not found',
-                ], 200);
+                ], StatusCode::$badRequest);
             }
             $notes = $validCategory->notes()->get();
-            
+
             return response()->json($notes, 200);
```
Comment (bao8816) [inline smart-note-server/app/Http/Controllers/NoteController.php:71]: 200 -> StatusCode

File: smart-note-server/app/Http/Controllers/NoteController.php [hunk 4]
```diff
@@ -93,21 +93,132 @@ public function createNote(CreateNoteRequest $request, string $category_id)
...
+                $fileName = StringValidation::replaceByPrefix(self::STORE_PATH_PREFIX, $attachmentPath);
+                $validNote->attachment = $fileName;
+            }
+
+            $validNote->save();
+
+            return response()->json([
+                'note' => $validNote,
+                'message' => 'Update new note successfully',
             ], 201);
```
Comment (bao8816) [inline smart-note-server/app/Http/Controllers/NoteController.php:215]: 201 -> StatusCode

File: smart-note-server/app/Http/Controllers/NoteController.php [hunk 5]
```diff
@@ -93,21 +93,132 @@ public function createNote(CreateNoteRequest $request, string $category_id)
...
         }
         catch (Exception $exception) {
             return response()->json([
                 'error' => $exception->getMessage(),
             ], 500);
```
Comment (bao8816) [inline smart-note-server/app/Http/Controllers/NoteController.php:220]: 500 -> StatusCode

**Review by bao8816 — COMMENTED** [review]
Check my comments in each file

**Review by bao8816 — APPROVED** [review]

**Pull request merged**